homeassistant/components/bluesound/media_player.py

```diff
@@ -449,6 +448,22 @@ async def async_update_status(self):
             _LOGGER.info("Client connection error, marking %s as offline", self._name)
             raise
 
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return f"{format_mac(self._sync_status.get('@mac'))}-{self.port}"
```

Comment:
`None` is not a valid mac address.

Reply:
How do we know that the port is stable?

Reply:
To my knowledge (and I have been testning the api and players alot) there is no way to change the port. If a player only has one zone it is always the default port. Has been for at least 5 years, I don’t see any reason for them changing it. 
of course it would be better not to use it, but I don’t see any way around it. 

As I explained earlier, they are them self using ip:port as a unique identifier. 

Screenshot from the documentation:
![7960058A-E08D-4B05-B644-35041A01DA7D](https://user-images.githubusercontent.com/5276026/196664409-141830ed-c695-4e22-940c-e303a93bc561.jpeg)

Reply:
Ok.

Reply:
@thrawnarn most of the code uses `get`, do we know which parts are required in the api?

Reply:
@fredrike not really. When I made the first implementation there were no api docs for it. Now I can’t see anything in the docs about required fields. But i would be very suprised if mac or port were missing in the response.